Whitelist more notifyd notifications for WebContent

bnham opened the pull request:
#### 957ddb2b94a123946d5ef1e6420bcbcceca2f53c
<pre>
Whitelist more notifyd notifications for WebContent
<a href="https://bugs.webkit.org/show_bug.cgi?id=274910">https://bugs.webkit.org/show_bug.cgi?id=274910</a>
<a href="https://rdar.apple.com/128970749">rdar://128970749</a>

Reviewed by Per Arne Vollan.

After looking at more data it looks like we are still not forwarding a number of notifyd
notifications that WebContent cares about when ENABLE_NOTIFY_BLOCKING is on.

Add in those missing notification names and also refactor the way the list of notifications is
handled, since that knowledge is currently distributed across 3 places in the codebase:

 - process-entitlements.sh
 - WebProcessPool::registerNotificationObservers()
 - iOS/macOS WebContentsandbox profile

Ideally these lists would come from some centralized file. But for now, I&apos;ve at least reorganized
things so it should be easier to keep these lists in sync.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/279548@main">https://commits.webkit.org/279548@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2a9a9d154335383cef9ed1c556f5d5d38d2b5a71

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53742 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33110 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6258 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57022 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4467 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40599 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4311 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43524 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2921 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55841 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/31323 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46466 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24660 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/53581 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28148 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2622 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49866 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3970 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58616 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28909 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4021 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50933 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30104 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46627 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50282 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11711 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31039 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29883 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -1475,24 +1478,24 @@
+        "com.apple.sessionagent.screenLockUIIsShown"
+        "com.apple.sessionagent.shieldWindowIsShowing"
+        "com.apple.sessionagent.shieldWindowLowered"
+        "com.apple.sessionagent.shieldWindowRaised"
```
Are these in use on iOS?

bnham replied:
Not sure, my guess is possibly not because it is coming from `_NSAccessibilitySafeToVendElement` which is in AppKit.

pvollan reviewed (approved):
Very nice, thank you! R=me.

bnham committed:
Whitelist more notifyd notifications for WebContent
https://bugs.webkit.org/show_bug.cgi?id=274910
rdar://128970749

Reviewed by Per Arne Vollan.

After looking at more data it looks like we are still not forwarding a number of notifyd
notifications that WebContent cares about when ENABLE_NOTIFY_BLOCKING is on.

Add in those missing notification names and also refactor the way the list of notifications is
handled, since that knowledge is currently distributed across 3 places in the codebase:

 - process-entitlements.sh
 - WebProcessPool::registerNotificationObservers()
 - iOS/macOS WebContentsandbox profile

Ideally these lists would come from some centralized file. But for now, I've at least reorganized
things so it should be easier to keep these lists in sync.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/279548@main
